bundler/lib/bundler/definition.rb

```diff
@@ -456,9 +456,9 @@ def ensure_equivalent_gemfile_and_lockfile(explicit_flag = false)
       @locked_deps.each  {|name, d| both_sources[name][1] = d.source }
 
       both_sources.each do |name, (dep, lock_source)|
-        next unless (dep.nil? && !lock_source.nil?) || (!dep.nil? && !lock_source.nil? && !lock_source.can_lock?(dep))
+        next if lock_source.nil? || (dep && lock_source.can_lock?(dep))
         gemfile_source_name = (dep && dep.source) || "no specified source"
-        lockfile_source_name = lock_source || "no specified source"
```

Comment:
If I understand this well, `lock_source` was already checked for presence before and this removed part was never reached thus it is ok to remove. Can you confirm?

Reply:
Yes, that's my reasoning. The previous `next` guard skips to the next iteration of `lock_source` is `nil`, so this could only be reached if `lock_source` was `false`. But `false` is not a valid value for a source.